src/openseadragon.js

```diff
@@ -2891,6 +2895,98 @@ function OpenSeadragon( options ){
         }
     }
 
+    /**
+     * Promise proxy in OpenSeadragon, enables $.supportsAsync feature.
+     * @type {PromiseConstructor}
+     */
+    $.Promise = window["Promise"] && $.supportsAsync ? window["Promise"] : class {
```

Comment:
Asynchronous pipeline is necessary only when you have async handlers (e.g. filtering plugin caman library usage) or async convertors (e.g. image onload). However, you might
 - want to test some behavior and the async pipeline just makes everything hard to test / debug / develop (because debugger jumps wildly)
 - want to fall back to the old synchronous behavior (backwards compatibility feature) because you for example rely on synchronous tile data update at render time (which is not possible as of now)

Reply:
Btw see your quesion: `Is there any way we can have a compatibility mode or plugin or something as a bridge for old projects and plugins?`

Reply:
Cool. Sounds like something that should go in the documentation somewhere.